flink-connectors/flink-connector-hbase-2.2/src/main/java/org/apache/flink/connector/hbase2/source/HBaseRowDataAsyncLookupFunction.java

```diff
@@ -0,0 +1,232 @@
+                        }
+                    }
+                }
+            });
+    }
+
+    private Configuration prepareRuntimeConfiguration() {
+        // create default configuration from current runtime env (`hbase-site.xml` in classpath) first,
+        // and overwrite configuration using serialized configuration from client-side env (`hbase-site.xml` in classpath).
+        // user params from client-side have the highest priority
+        Configuration runtimeConfig = HBaseConfigurationUtil.deserializeConfiguration(
+                serializedConfig,
+                HBaseConfigurationUtil.getHBaseConfiguration());
+
+        // do validation: check key option(s) in final runtime configuration
+        if (StringUtils.isNullOrWhitespaceOnly(runtimeConfig.get(HConstants.ZOOKEEPER_QUORUM))) {
+            LOG.error("can not connect to HBase without {} configuration", HConstants.ZOOKEEPER_QUORUM);
+            throw new IllegalArgumentException("check HBase configuration failed, lost: '" + HConstants.ZOOKEEPER_QUORUM + "'!");
+        }
+
+        return runtimeConfig;
+    }
+
+    @Override
+    public void open(FunctionContext context) {
+        LOG.info("start open ...");
+        Configuration config = prepareRuntimeConfiguration();
+        CompletableFuture<AsyncConnection> asyncConnectionFuture = ConnectionFactory.createAsyncConnection(config);
+        try {
+            asyncConnection = asyncConnectionFuture.get();
+            table = asyncConnection.getTable(TableName.valueOf(hTableName), (ExecutorService) Executors.directExecutor());
+
+            this.cache = cacheMaxSize == -1 || cacheExpireMs == 0 ? null : CacheBuilder.newBuilder()
+                    .recordStats()
+                    .expireAfterWrite(cacheExpireMs, TimeUnit.MILLISECONDS)
+                    .maximumSize(cacheMaxSize)
+                    .build();
+            if (cache != null) {
+                context.getMetricGroup().gauge("lookupCacheHitRate", (Gauge<Double>) () -> cache.stats().hitRate());
+            }
+        } catch (InterruptedException | ExecutionException e) {
+            LOG.error("Exception while creating connection to HBase.", e);
+            throw new RuntimeException("Cannot create connection to HBase.", e);
+        }
+        this.serde = new HBaseSerde(hbaseTableSchema, nullStringLiteral);
+        LOG.info("end open.");
+    }
+
+    @Override
+    public void close() {
+        LOG.info("start close ...");
+        if (null != asyncConnection) {
+            try {
+                asyncConnection.close();
+                asyncConnection = null;
+            } catch (IOException e) {
+                // ignore exception when close.
+                LOG.warn("exception when close connection", e);
+            }
+        }
+        LOG.info("end close.");
+    }
+
+    @VisibleForTesting
+    public String getHTableName() {
+        return hTableName;
+    }
+
+
+}
```

Comment:
I think we should add unit test for this class just like `JdbcRowDataLookupFunctionTest` and a ITCase in HBaseConnectorITCase as well